libbpfgo.go

```diff
@@ -442,21 +442,22 @@ func (b *BPFMap) GetMaxEntries() uint32 {
 
 func GetUnsafePointer(data interface{}) (unsafe.Pointer, error) {
 	var dataPtr unsafe.Pointer
-	if k, isType := data.(int8); isType {
+	switch k := data.(type) {
```

Comment:
Nice!
I wasn't aware of this type switch option.
And now I'm asking myself - do we even need the switch-case?
All the cases do the same thing - `dataPtr = unsafe.Pointer(&k)`
maybe just fallthrough?

Reply:
Ok, after reading about type switches I understand that we must do it this way so that k will have the correct type asserted.

Reply:
Yeah, this could actually be refactored further into something like:

```
switch k := data.(type) {
case <all ints>: ....
case []byte: ....
default: ....
```

However I'm curious why this function is so restrictive in the first place. I've actually spoken with @grantseltzer offline about this and plan on hacking on this further in order to allow more arbitrary types, if possible.

Reply:
When I wrote this code, I tried to find a generic method to get a pointer to the concrete type so I can pass it to libbpf, but couldn't find any. So I just coded the types that were needed for tracee, which I guess are the most common ones.
I'll be happy to change this code to something more generic if there is a sple way to do that

Reply:
Ack, makes sense!